### ISSUE TITLE
Fail the build on changing the content of a changelog

### ISSUE THREAD

**Issue by pradipta** [cloudyrock/mongock#275]
Had used Liquibase with MySQL previously. With Liquibase, on changing the content of an already executed changelog fails the build. Which seems legit for obvious reasons.
However, that probably is not the case with Mongock. Correct me if I am wrong.
Do we want something like that?

**Comment by dieppa** [comment]
Hi @pradipta ,

if I am not wrong, liquibase uses the checksum to check if an existing changeLog has been modified. From my point of view I see pros and cons. I didn't have a good experience with that feature when I used Liquibase, I found it a bit cumbersome. But I can see the pros too. 

But I like Mongock to be built from the user's point of view, so what are your thoughts about?

**Comment by pradipta** [comment]
Totally understand your point. Even I have faced that and fixing those builds are a pain.
But it does provide some safety and convenience.
This can probably be a property like `restrict-changeset-modification`
Totally depends on your and other contributors' and users' opinion.

**Comment by dieppa** [comment]
Yeah, it seems that can be added optionally by properties, as everything we do.

Good suggestion, thanks! I will add it to the road map ;)

**Comment by pradipta** [comment]
However, this could lead to some issues.
Consider this changeset:
```
@ChangeSet(order = "001", id = "insert user", author = "pradipta")
public void changeWithMongockTemplate(MongockTemplate mongockTemplate) {
  User user = new User("name", "email"); //User has two fields, name and email
  mongockTemplate.save(myEntity);
}
```

Now, if I add a new field to User, lets say `String phoneNumber`.
I would now have to change the method to add the new field to avoid compilation failure.

**Comment by dieppa** [comment]
Hello @pradipta ,

yes, that's one of the issue when using domain classes. You need to change the changeSet, which is not ideal, plus the checksum validation would break. 

There are different ways of avoiding this. From not using domain classes to delegate on another external classes, called factory if you wish, the creation of the object, passing through keeping the old constructor, with the two field.

That depends on your design. We'll work on a "best practice" or recommendation for this kind of situation
